app/src/main/java/org/mozilla/fenix/tabstray/viewholders/NormalBrowserPageViewHolder.kt

```diff
@@ -79,11 +79,12 @@ class NormalBrowserPageViewHolder(
         val browserAdapter = concatAdapter.browserAdapter
         val inactiveTabAdapter = concatAdapter.inactiveTabsAdapter
         val tabGroupAdapter = concatAdapter.tabGroupAdapter
+        val inactiveTabsAreEnabled = containerView.context.settings().inactiveTabsAreEnabled
 
         val selectedTab = browserStore.state.selectedNormalTab ?: return
 
         // Update tabs into the inactive adapter.
-        if (FeatureFlags.inactiveTabs && selectedTab.isNormalTabInactive(maxActiveTime)) {
+        if (FeatureFlags.inactiveTabs && selectedTab.isNormalTabInactive(maxActiveTime) && inactiveTabsAreEnabled) {
```

Comment:
The check for the flag here is not needed.

Reply:
Good catch